example/lambdas/db-provision-user-database/src/index.ts

```diff
@@ -15,42 +17,28 @@ const validateDatabaseInput = (
   inputString: string, regexp: RegExp
 ): boolean => regexp.test(inputString);
 
-const getConnectionConfig = async (SecretId: string): Promise<Knex.PgConnectionConfig> => {
-  const secretsManager = new AWS.SecretsManager();
-  const response = await secretsManager.getSecretValue(
-    { SecretId } as AWS.SecretsManager.GetSecretValueRequest
-  ).promise();
-  if (response.SecretString === undefined) {
-    throw new Error('Database credentials are undefined!');
-  }
-  const dbAccessMeta = JSON.parse(response.SecretString);
-  return {
-    host: dbAccessMeta.host,
-    user: dbAccessMeta.username,
-    password: dbAccessMeta.password,
-  };
-};
-
 export const tableExists = async (tableName: string, knex: Knex) =>
   knex('pg_database').select('datname').where(knex.raw(`datname = CAST('${tableName}' as name)`));
 
 export const userExists = async (userName: string, knex: Knex) =>
   knex('pg_catalog.pg_user').where(knex.raw(`usename = CAST('${userName}' as name)`));
 
+const validateEvent = (event: HandlerEvent): void => {
+  if (event?.dbPassword === undefined || event?.prefix === undefined) {
+    throw new Error(`This lambda requires 'dbPassword' and 'prefix' to be defined on the event: ${event}`);
+  }
+};
+
 export const handler = async (event: HandlerEvent): Promise<void> => {
+  validateEvent(event);
   let knex;
-
   try {
-    const secretsManager = new AWS.SecretsManager();
-    const config = await getConnectionConfig(event.rootLoginSecret);
-    knex = Knex({
-      client: 'pg',
-      connection: config,
-      acquireConnectionTimeout: 120000,
-    });
-    const dbUser = event?.prefix.replace(/-/g, '_');
+    knex = await connection.knex(
+      { databaseCredentialSecretArn: event.rootLoginSecret }
+    );
+    const dbUser = event.prefix.replace(/-/g, '_');
 
-    [dbUser, event?.dbPassword].forEach((input) => {
+    [dbUser, event.dbPassword].forEach((input) => {
       if (!(validateDatabaseInput(input, new RegExp(/^\w+$/)))) {
```

Comment:
I realize that it isn't part of this PR, but I'm not sure we need a separate function that just turns around and calls `.test`.